src/com/ms/silverking/thread/lwt/LWTThread.java

```diff
@@ -13,267 +13,264 @@
+          } catch (Exception e) {
+            Log.logErrorWarning(e);
+            ThreadUtil.pauseAfterException();
+          }
         } else {
-            runMultiple();
+          idleLock.lock();
+          try {
+            idleCV.await();
+          } finally {
+            idleLock.unlock();
+          }
         }
-    }
-    
-    private void runSingle() {
-        ThreadState.setLWTThread();
-        while (running) {
-            try {
-                while (active) {
-                    AssignedWork    work;
-                    
-                    work = q.take();
-                    if (Log.levelMet(Level.FINE)) {
-                        Log.fine(this +" doWork "+ work);
-                    }
-                    //System.out.println(idleThreads.get());
-                    try {
-                        work.doWork();
-                    } catch (Exception e) {
-                        Log.logErrorWarning(e);
-                        ThreadUtil.pauseAfterException();                
-                    }
-                }
-                while (!active) {
-                    idleLock.lock();
-                    try {
-                        idleCV.await();
-                    } finally {
-                        idleLock.unlock();
-                    }
-                }
-            } catch (Exception e) {
-                Throwable    t;
-                
-                Log.warning("************************************** "+ getName());
-                t = e;
-                while (t != null) {
-                    Log.logErrorWarning(t);
-                    t = t.getCause();
-                    Log.warning("......................................");
-                }
-                ThreadUtil.pauseAfterException();
-            }
+      } catch (Exception e) {
+        Throwable t;
+
+        Log.warning("************************************** " + getName());
+        t = e;
+        while (t != null) {
+          Log.logErrorWarning(t);
+          t = t.getCause();
+          Log.warning("......................................");
         }
+        ThreadUtil.pauseAfterException();
+      }
     }
-    
-    private void activeMultiple(AssignedWork[] workList) {
-        BaseWorker  worker;
-        int         numWorkItems;
-        
-        try {
-            numWorkItems = q.takeMultiple(workList);
+  }
+
+  private void activeMultiple(AssignedWork[] workList) {
+    BaseWorker worker;
+    int numWorkItems;
+
+    try {
+      numWorkItems = q.takeMultiple(workList);
             /*
             AssignedWork    w;
             
             w = q.take();
             workList[0] = w;
             numWorkItems = 1;
             */
-        } catch (InterruptedException ie) {
-            throw new RuntimeException("Interruption not supported", ie);
-        }
-        if (enableStats) {
-            ++workUnitStats[numWorkItems];                    
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)